src/content/peerconnection/constraints/js/main.js

```diff
@@ -202,6 +203,43 @@ function onAddIceCandidateError(error) {
   trace('Failed to add Ice Candidate: ' + error.toString());
 }
 
+function videoStatistics(stream, video) {
+  var sizeFromTag;
+  var sizeFromSettings;
+  var frameRate;
+  // Collect some stats from the video tags.
+  if (video.videoWidth) {
+    sizeFromTag = video.videoWidth + 'x' + video.videoHeight;
+  }
+  // Use track's info if available.
+  // NOTE: getSettings is under development, so code adapts to various
+  // in-flight situations that aren't according to spec.
+  if (stream) {
+    if (stream.getVideoTracks()[0]) {
+      if (stream.getVideoTracks()[0].getSettings) {
+        var settings = stream.getVideoTracks()[0].getSettings();
```

Comment:
I can do that - I've found it useful in other cases. The global object should probably be videoSettings. (apologies for being slow)